### PR TITLE
[EP Partitioning Utils] Add check for assigned node.

### PR DESCRIPTION
**Pull request by edgchen1**
**Description**
Adds a check that a node is not already assigned to an EP before adding it to an EP partition.

**Motivation and Context**
This check should be done by EP partitioning.

### CODE REVIEW

**Review by gwang-msft — DISMISSED** [review]

File: onnxruntime/core/providers/partitioning_utils.cc [hunk 1]
```diff
@@ -156,7 +156,9 @@ std::vector<std::vector<const Node*>> CreateSupportedPartitionNodeGroups(
     const Node& node = *nodes_to_process.front();
     nodes_to_process.pop_front();
 
-    const bool is_node_supported = is_node_supported_fn(node);
+    const bool is_node_supported =
+        node.GetExecutionProviderType().empty() &&  // a node that is already assigned to an EP is unsupported
```
Comment (gwang-msft) [inline onnxruntime/core/providers/partitioning_utils.cc:160]: In this case, maybe log some info out? I assume this should not happen

Comment (edgchen1) [reply onnxruntime/core/providers/partitioning_utils.cc:160]: an existing value from GetExecutionProviderType() is not unexpected, for example, if another EP has already taken the node

**Review by skottmckay — APPROVED** [review]
:shipit:

**Review by skottmckay — APPROVED** [review]
:shipit:

**Pull request merged**